### PR TITLE
pdpb: add ScanRegions

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### CODE REVIEW

File: proto/pdpb.proto [hunk 1]
```diff
@@ -204,6 +206,17 @@ message GetRegionByIDRequest {
 
 // Use GetRegionResponse as the response of GetRegionByIDRequest.
 
+message ScanRegionsRequest {
+    RequestHeader header = 1;
+    bytes region_key = 2;
```
Comment (MyonKeminta) [inline proto/pdpb.proto]: It seems that there should be an empty line above here?

Comment (disksing) [reply proto/pdpb.proto]: ok

File: proto/pdpb.proto [hunk 2]
```diff
@@ -204,6 +206,19 @@ message GetRegionByIDRequest {
...
+    RequestHeader header = 1;
+
+    bytes region_key = 2;
+    int32 limit = 3;
+}
+
+message ScanRegionsResponse {
+    ResponseHeader header = 1;
+    repeated metapb.Region regions = 2;
+    repeated metapb.Peer leaders = 3;
```
Comment (zhangjinpeng1987) [inline proto/pdpb.proto]: How about region has no leader?

File: proto/pdpb.proto (hunk 1 above)
Comment (zhangjinpeng1987) [reply proto/pdpb.proto]: I think `start_key` is a better name.

Comment (disksing) [reply proto/pdpb.proto]: ok

File: proto/pdpb.proto (hunk 2 above)
Comment (disksing) [reply proto/pdpb.proto]: It will add empty value to make regions and leaders are matched.

File: proto/pdpb.proto [hunk 3]
```diff
@@ -204,6 +206,19 @@ message GetRegionByIDRequest {
...
+message ScanRegionsRequest {
+    RequestHeader header = 1;
+
+    bytes start_key = 2;
+    int32 limit = 3;
+}
+
+message ScanRegionsResponse {
+    ResponseHeader header = 1;
+    repeated metapb.Region regions = 2;
```
Comment (MyonKeminta) [inline proto/pdpb.proto]: Oh, here should be an empty line too

**Comment by zhangjinpeng1987** [comment]
@disksing Please cherry-pick to release-3.0 branch, thx.

**Pull request merged**